### DISCUSSION TITLE
Headless Listbox crashing with minor customisation

### DISCUSSION THREAD

**Discussion by jardakotesovec** [alpinejs/alpine#4448] [7. Bugs (archived)] [answered]
Would like to use headless listbox. Unfortunately very minor customisation is causing it to crash.

Its exactly the same as the example on the website (https://alpinejs.dev/component/headless-listbox), only difference is adding that getter filteredFrameworks to remove currently selected framework from the list of options.

Here is the reproduction - https://jsfiddle.net/0fanjg27/ .

Any suggestion how to work around it or ideally fix that bug? 

<img width="551" alt="Screenshot 2024-11-21 at 14 11 06" src="https://github.com/user-attachments/assets/00f67af8-ee99-4c7d-a256-2bb46933810a">

**Comment by ekwoka** [comment, marked as answer]
Basically, since you're filtering like that, the element won't be in the listbox anymore, so it won't have the refs.

I'd recommend for such a case, to just use an x-show on the options.

**Reply by jardakotesovec** [reply]
@ekwoka Hi, thanks for response.

If I select something I would expect that focus would be triggered on the the listbox button (x-listbox:button), not on the item (x-listbox:option).  Even the name refs suggests its triggering focus on the button.

Am I missing something here?

**Reply by ekwoka** [reply]
But the focus is triggered FROM the listbox option is the issue.

It does it on the next tick, which results in that element being removed and cleaned up (mostly) before it runs.

Not exactly sure why it does it that way, but it does.

**Reply by jardakotesovec** [reply]
@ekwoka Alright, thanks a lot for help!
